File: app/src/main/java/com/donald/musictheoryapp/Utils/ProgressBarOnTouchListener.java
package com.donald.musictheoryapp.Utils;

import android.view.MotionEvent;
import android.view.View;
import android.widget.SeekBar;

import com.donald.musictheoryapp.Music.MusicXML.Clef;
import com.donald.musictheoryapp.Music.MusicXML.Note;

public class ProgressBarOnTouchListener implements View.OnTouchListener
{
	private final SeekBar seekBar;
	private float initialX;
	private int initialProgress;

	public ProgressBarOnTouchListener(SeekBar seekBar)
	{
		this.seekBar = seekBar;
	}

	@Override
	public boolean onTouch(View view, MotionEvent event)
	{
		// TODO: IMPLEMENT APPLE-STYLE DRAG, FURTHER UP SLOWS DOWN DRAG
		int action = event.getActionMasked();
		switch(action)
		{
			case MotionEvent.ACTION_DOWN:
			{
				initialX = event.getX();
				initialProgress = seekBar.getProgress();
				return true;
			}
			case MotionEvent.ACTION_MOVE:
			{
				float currentX = event.getX();
				float distance = currentX - initialX;
				int newProgress =
					Math.min(
						Math.max(
							initialProgress + (int) (distance * 2.5f / (seekBar.getMax() - seekBar.getMin())),
							seekBar.getMin()
						),
						seekBar.getMax()
					);
				seekBar.setProgress(newProgress);
				return true;
			}
			case MotionEvent.ACTION_UP:
			{
				return true;
			}
			default:
			{
				return view.performClick();
			}
		}
	}
}

File: app/src/main/java/com/donald/musictheoryapp/listener/ProgressBarOnTouchListener.kt
package com.donald.musictheoryapp.listener

import android.widget.SeekBar
import android.view.View.OnTouchListener
import android.view.MotionEvent
import android.view.View

class ProgressBarOnTouchListener(private val seekBar: SeekBar) : OnTouchListener {

    private var initialX = 0f
    private var initialProgress = 0

    override fun onTouch(view: View, event: MotionEvent): Boolean {
        // TODO: IMPLEMENT APPLE-STYLE DRAG, FURTHER UP SLOWS DOWN DRAG
        return when (event.actionMasked) {
            MotionEvent.ACTION_DOWN -> {
                initialX = event.x
                initialProgress = seekBar.progress
                true
            }
            MotionEvent.ACTION_MOVE -> {
                val currentX = event.x
                val distance = currentX - initialX
                val newProgress = (initialProgress + (distance * 2.5f / (seekBar.max)).toInt()).coerceAtLeast(0).coerceAtMost(seekBar.max)
                seekBar.progress = newProgress
                true
            }
            MotionEvent.ACTION_UP -> {
                true
            }
            else -> {
                view.performClick()
            }
        }
    }

}